packages/solana-swap-find-provider/lib/SolanaSwapFindProvider.ts

```diff
@@ -0,0 +1,147 @@
+import { BigNumber, SwapParams, SwapProvider, Transaction } from '@liquality/types'
+import { Provider } from '@liquality/provider'
+import { PendingTxError } from '@liquality/errors'
+import { sha256 } from '@liquality/crypto'
+
+import _filter from 'lodash/filter'
+
+export default class SolanaSwapFindProvider extends Provider implements Partial<SwapProvider> {
+  private instructions = {
+    init: 0,
+    claim: 1,
+    refund: 2
+  }
+
+  async findInitiateSwapTransaction(swapParams: SwapParams): Promise<Transaction<any>> {
+    const { refundAddress } = swapParams
+
+    return await this._findTransactionByAddress({
+      address: refundAddress as string,
+      swapParams,
+      instruction: this.instructions.init,
+      validation: this._compareParams
+    })
+  }
+
+  async findClaimSwapTransaction(swapParams: SwapParams, initiationTxHash: string): Promise<Transaction<any>> {
+    const [initTransaction] = await this.getMethod('getParsedAndConfirmedTransactions')([initiationTxHash])
```

Comment:
You should use the standard interface which is provided by all chains (e.g. getTransactionReceipt). 

`getParsedAndConfirmedTransactions` is Solana specific. You can use it internally (e.g. inside `getTransactionReceipt`)